packages/video_player/video_player/lib/video_player.dart

```diff
@@ -375,6 +374,9 @@ class VideoPlayerController extends ValueNotifier<VideoPlayerValue> {
   /// has been sent to the platform, not when playback itself is totally
   /// finished.
   Future<void> play() async {
+    if (value.position == value.duration) {
+      await seekTo(const Duration());
+    }
```

Comment:
If video is at the end, play() restarts the video from the beginning. This isn't necessary to fix the original issue, but it makes the behavior on iOS and Android consistent with web.